ci/test.go

```diff
@@ -0,0 +1,191 @@
+		Sync(ctx)
+	return err
+}
+
+func (t *Test) testCmd(ctx context.Context) (*Container, error) {
+	engine := t.Dagger.Engine().
+		WithConfig(`registry."registry:5000"`, `http = true`).
+		WithConfig(`registry."privateregistry:5000"`, `http = true`).
+		WithConfig(`registry."docker.io"`, `mirrors = ["mirror.gcr.io"]`).
+		WithConfig(`grpc`, `address=["unix:///var/run/buildkit/buildkitd.sock", "tcp://0.0.0.0:1234"]`).
+		WithArg(`network-name`, `dagger-dev`).
+		WithArg(`network-cidr`, `10.88.0.0/16`)
+	devEngine, err := engine.Container(ctx, "")
+	if err != nil {
+		return nil, err
+	}
+
+	devBinary, err := t.Dagger.CLI().File(ctx, "")
+	if err != nil {
+		return nil, err
+	}
+
+	// This creates an engine.tar container file that can be used by the integration tests.
+	// In particular, it is used by core/integration/remotecache_test.go to create a
+	// dev engine that can be used to test remote caching.
+	// I also load the dagger binary, so that the remote cache tests can use it to
+	// run dagger queries.
+
+	// These are used by core/integration/remotecache_test.go
+	testEngineUtils := dag.Directory().
+		WithFile("engine.tar", devEngine.AsTarball()).
+		WithFile("dagger", devBinary, DirectoryWithFileOpts{
+			Permissions: 0755,
+		})
+
+	registrySvc := registry()
+	devEngineSvc := devEngine.
+		WithServiceBinding("registry", registrySvc).
+		WithServiceBinding("privateregistry", privateRegistry()).
+		WithExposedPort(1234, ContainerWithExposedPortOpts{Protocol: Tcp}).
+		WithMountedCache(distconsts.EngineDefaultStateDir, dag.CacheVolume("dagger-dev-engine-test-state"+identity.NewID())).
+		WithExec(nil, ContainerWithExecOpts{
+			InsecureRootCapabilities: true,
+		}).
+		AsService()
+
+	endpoint, err := devEngineSvc.Endpoint(ctx, ServiceEndpointOpts{Port: 1234, Scheme: "tcp"})
+	if err != nil {
+		return nil, err
+	}
+
+	cliBinPath := "/.dagger-cli"
+
+	utilDirPath := "/dagger-dev"
+	tests := util.GoBase(t.Dagger.Source).
+		WithExec([]string{"go", "install", "gotest.tools/gotestsum@v1.10.0"}).
+		WithMountedDirectory("/app", t.Dagger.Source). // need all the source for extension tests
+		WithMountedDirectory(utilDirPath, testEngineUtils).
+		WithEnvVariable("_DAGGER_TESTS_ENGINE_TAR", filepath.Join(utilDirPath, "engine.tar")).
+		WithWorkdir("/app").
+		WithServiceBinding("dagger-engine", devEngineSvc).
+		WithServiceBinding("registry", registrySvc)
+
+	if t.CacheConfig != "" {
+		tests = tests.WithEnvVariable("_EXPERIMENTAL_DAGGER_CACHE_CONFIG", t.CacheConfig)
+	}
+
+	return tests.
+			WithMountedFile(cliBinPath, devBinary).
+			WithEnvVariable("_EXPERIMENTAL_DAGGER_CLI_BIN", cliBinPath).
+			WithEnvVariable("_EXPERIMENTAL_DAGGER_RUNNER_HOST", endpoint),
+		// TODO: re-enable this, to avoid rate-limiting
+		// introduced in https://github.com/dagger/dagger/commit/20241ba7a8d39da8f6ff185d512953b6b0ec1a55
+		// WithMountedDirectory("/root/.docker", util.HostDockerDir(dag)),
```

Comment:
Oh you know what, I think I remember now why this was a thing.

As mentioned in the discord thread, in theory the auth session attachable should handle this. But the problem is we are creating a nested exec and currently nested execs end up as their own completely isolated session, so they don't get that auth session attachable.

Fortunately, that will no longer be the case after https://github.com/dagger/dagger/pull/6916 lands. So I think we can just rm these comments; it's not possible to implement this now and the need to should disappear soon.

cc @vito since you were on that discord thread too, just in case you're curious

Reply:
Uh oh, I've had a realization as to why we had it like this :laughing: 

https://github.com/dagger/dagger/actions/runs/8603708206/job/23576176153

Since we spin up nested engines (for things like code gen / sdk tests / etc), we need a way of passing our token through. We seem to have hit this *pretty* quickly, so we either need #6916 to land soon, or we need to chuck in this little hack again, which is sad.